fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/ConvertToEqualForNullRule.java

```diff
@@ -0,0 +1,113 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.sql.optimizer.rule.transformation;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Lists;
+import com.starrocks.analysis.BinaryType;
+import com.starrocks.sql.optimizer.OptExpression;
+import com.starrocks.sql.optimizer.OptimizerContext;
+import com.starrocks.sql.optimizer.Utils;
+import com.starrocks.sql.optimizer.operator.OperatorType;
+import com.starrocks.sql.optimizer.operator.logical.LogicalJoinOperator;
+import com.starrocks.sql.optimizer.operator.pattern.Pattern;
+import com.starrocks.sql.optimizer.operator.scalar.BinaryPredicateOperator;
+import com.starrocks.sql.optimizer.operator.scalar.CompoundPredicateOperator;
+import com.starrocks.sql.optimizer.operator.scalar.IsNullPredicateOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ScalarOperator;
+import com.starrocks.sql.optimizer.rule.RuleType;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+public class ConvertToEqualForNullRule extends TransformationRule {
+    public ConvertToEqualForNullRule() {
+        super(RuleType.TF_CONVERT_TO_EQUAL_FOR_NULL_RULE, Pattern.create(OperatorType.LOGICAL_JOIN).
+                addChildren(Pattern.create(OperatorType.PATTERN_LEAF), Pattern.create(OperatorType.PATTERN_LEAF)));
```

Comment:
should be a common expression rewrite？not only join

Reply:
Rewriting it in join on condition can achieve good benefits. I am worried about it may affect the efficiency of common expression rewriting process, because its usage scenarios may be relatively few?